ingestion-edge/ingestion_edge/publish.py

```diff
@@ -25,6 +25,37 @@
 )
 
 
+def _legacy_encoding(value: Optional[str], request: Request) -> Optional[str]:
+    """Convert string from UTF-8 with surrogate escape characters to ISO-8859-1.
+
+    As of Sanic 21.3.0 in sanic.http.http1 headers are decoded with the default codec
+    (UTF-8) and errors="surrogateescape", but behavior prior to that was to decode
+    headers with ISO-8859-1, aka latin-1.
+
+    Unit tests hit decode headers in sanic.http.http1, but prod and some integration
```

Comment:
If I'm reading this right, we're changing the behavior in prod because of the unit tests - is that right? Is it possible that we change the tests instead and leave prod as-is?